### PR TITLE
add new endpoint to fetch node features

### PR DESCRIPTION
**Pull request by Eslam-Nawara**
### Description

add new endpoint to fetch node features

### Changes

add new endpoint to fetch node features

### Related Issues

- #2421 

### Checklist

- [ ] Tests included
- [x] Build pass
- [x] Documentation
- [x] Code format and docstring

### CODE REVIEW

File: client/node.go [hunk 1]
```diff
@@ -322,6 +320,13 @@ func (n *NodeClient) NetworkGetPublicConfig(ctx context.Context) (cfg pkg.Public
 	return
 }
 
+func (n *NodeClient) NetworkGetNodeFeatures(ctx context.Context) (feat pkg.NodeFeatures, err error) {
+	const cmd = "zos.network.node_features_get"
```
Comment (xmonader) [inline client/node.go]: Why on the zos.network? I assumed it will be on `zos.system.node_features`

Comment (Eslam-Nawara) [reply client/node.go]: since this is related to netowrkd-light I assumed it would be on network, but can move it to be system.node_features

Comment (xmonader) [reply client/node.go]: My reasoning is as a user I should ask the node what version do you have? (which is `zos.system.version`) or what features are enabled on the node so `zos.system.node_features_get`

File: pkg/network.go [hunk 2]
```diff
@@ -257,6 +260,32 @@ type PublicConfig struct {
 	Domain string `json:"domain"`
 }
 
+type NodeFeatures struct {
```
Comment (xmonader) [inline pkg/network.go]: NodeFeatures should have been a slice of a Feature []Feature and that feature is a typed string that can be any of `zdb, zmachine, .. etc)

File: pkg/monitor.go [hunk 3]
```diff
@@ -78,13 +78,17 @@ type PoolStats struct {
...
+type NodeFeature string
+
 // SystemMonitor interface (provided by noded)
 type SystemMonitor interface {
 	NodeID() uint32
 	Memory(ctx context.Context) <-chan VirtualMemoryStat
 	CPU(ctx context.Context) <-chan TimesStat
 	Disks(ctx context.Context) <-chan DisksIOCountersStat
 	Nics(ctx context.Context) <-chan NicsIOCounterStat
+	GetNodeFeatures() ([]NodeFeature, error)
```
Comment (xmonader) [inline pkg/monitor.go]: About the suggested interface, will this ever fail? worst case would be an empty slice :)

**Review by xmonader — APPROVED** [review]

**Review by muhamadazmy — APPROVED** [review]

**Pull request merged**